Modules/Quotas/src/Quotas.Application/DTOs/TierDetailsDTO.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Backbone.Modules.Quotas.Domain.Aggregates.Metrics;
+using Backbone.Modules.Quotas.Domain.Aggregates.Tiers;
+
+namespace Backbone.Modules.Quotas.Application.DTOs;
+public class TierDetailsDTO
+{
+    public TierDetailsDTO(Tier tier, IEnumerable<Metric> metrics)
+    {
+        Id = tier.Id;
+        Name = tier.Name;
+        Quotas = tier.Quotas.Select(quota => new TierQuotaDefinitionDTO(quota.Id,
+            new MetricDTO(quota.MetricKey, metrics.First(metric => metric.Key == quota.MetricKey).DisplayName),
+            quota.Max, quota.Period));
```

Comment:
```suggestion
        Quotas = tier.Quotas.Select(quota => 
            new TierQuotaDefinitionDTO(
                quota.Id,
                new MetricDTO(metrics.First(metric => metric.Key == quota.MetricKey)),
                quota.Max, 
                quota.Period
            )
        );
```

Reply:
Beside the formatting, the constructor of a MetricDTO still requires both a key and a display name, so it should still take those two like before right? Or should a new constructor be added to the MetricDTO that takes a metric as a parameter?

Reply:
The latter. If the other constructor is actually used somewhere, you can pass a metric there as well.

And you should probably add a default constructor for AutoMapper